axlearn/common/flash_attention/layer.py

```diff
@@ -190,11 +205,14 @@ def _compute_attention(
                 )
             attention_logit_biases = attention_logit_biases.astype(q_proj.dtype)
 
-        mask_fn = self._mask_fn
+        if attention_logit_biases is None or self._mask_fn is causal_mask:
```

Comment:
@changlan  I guess this condition should be 

if attention_logit_bias is None or not self._is_mask_fn_used()

Reply:
With today's `_logit_biases_for_mask` impl, it should be safe to just `if attention_logit_bias is None`.